messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -242,6 +246,31 @@ private boolean unsubscribe(String queryName,
                                                     updateHandlerRegistration.getRegistration());
     }
 
+    public <Q, I, U> SubscriptionQueryResult<QueryResponseMessage<I>, SubscriptionQueryUpdateMessage<U>> subscriptionQuery(
+            SubscriptionQueryMessage<Q, I, U> query,
+            int updateBufferSize) {
+
+        if (queryUpdateEmitter.queryUpdateHandlerRegistered(query)) {
+            throw new IllegalArgumentException("There is already a subscription with the given message identifier");
+        }
+
+        MonoWrapper<QueryResponseMessage<I>> initialResult = MonoWrapper.create(monoSink -> query(query)
+                .thenAccept(monoSink::success)
+                .exceptionally(t -> {
+                    logger.error(format("An error happened while trying to report an initial result. Query: %s", query),
+                                 t);
+                    monoSink.error(t.getCause());
+                    return null;
+                }));
+
+        UpdateHandlerRegistration<U> updateHandlerRegistration = queryUpdateEmitter
+                .registerUpdateHandler(query, updateBufferSize);
```

Comment:
Since the only difference between this and deprecated method is invocation on `queryUpdateEmitter` we could extract common parts into a private method and parameterize it with this specific invocation...

Reply:
we can, but I was hopping that we will just remove deprecated method in next release, so extraction might not be needed in this case...